Volume/Analytics/Analytics.swift

```diff
@@ -9,11 +9,82 @@
 import AppDevAnalytics
 
 // volume-specific extensions of base Event protocol
-
 struct StartOnboarding: Event {
     let name = "start_onboarding"
 }
 
 struct CompleteOnboarding: Event {
     let name = "complete_onboarding"
 }
+
+struct FollowPublication: Event {
+    let name = "follow_publication"
+    var parameters: [String : Any]?
```

Comment:
You define this exact field 8 different times. To reduce repetition, I would create a base protocol with this value. For example (with a better name)

```
protocol BaseEvent: Event {
    var parameters: [String: Any]?
}
```

Also, do we do `[String : Any]` or `[String: Any]` throughout the codebase? It should be consistent. If I recall correctly, Google's style guide has the latter, but you should confirm if you're unsure.

Reply:
Actually I change my mind. I don't think it's possible to omit the field name here unless you add extra code, which defeats the purpose. I would take a different approach:

```
struct AnyEvent: Event {
    let name: String
    let parameters: [String: Any]? 
    // make these vars if necessary 
}

enum VolumeEvent { // or whatever
    case startOnboarding = "start_onboarding"
    // ...

    func toEvent(parameters: [...]) {
        AnyEvent(name: rawValue, parameters: parameters)
    }
}
```